libra/libra
# [move] Implement authentication rule for Transaction::WriteSet.

## Motivation

Implemented the authentication rule for Transaction::WriteSet. The basic idea is that we add a new module called `LibraWriteSetManager` that contains the prologue/epilogue code to determine whether a WriteSet should be accepted or not. Due to the fact that the system is going to reconfigure after executing this transaction, `LibraVM` itself doesn't need to reload its configs nor need to cleanup its cache.

The code itself right now is only functional complete and probably require major code cleanup. I have a few remaining questions as well:
1. Do we need any rules on the submitted `ChangeSet`? Say no delete op as we used to?
2. What's the error handling policy for VM internal failures while processing those transactions?
3. Should VMValidator reject WriteSet with too new sequence number?

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

Added tests for following cases:
1. Execute and verify a well formed WriteSet transaction.
2. Execute and verify WriteSet transaction with bad sequence number/auth key.
3. Execute WriteSet transaction with bad contents (Namely writeset collides with events/writes generated by epilogue)
